jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/jsonb/JsonbDiagnosticsCollector.java

```diff
@@ -124,4 +137,34 @@ private void addJsonbTransientSpecificDiagnostics(Diagnostic d) {
         d.setMessage(JsonbConstants.ERROR_MESSAGE_JSONB_TRANSIENT);
         d.setCode(JsonbConstants.DIAGNOSTIC_CODE_ANNOTATION_TRANSIENT_FIELD);
     }
+    
+    private boolean hasJsonbTransient(IAnnotatable annotable) throws JavaModelException {
+        for (IAnnotation annotation : annotable.getAnnotations()) {
+            String annotationName = annotation.getElementName();
+            if (annotationName.equals(JsonbConstants.JSONB_TRANSIENT))
+                return true;
+        }
+        return false;
+    }
+    
+    private boolean hasNonJsonbTransientAnnotation(IAnnotatable annotable) throws JavaModelException {
+        for (IAnnotation annotation : annotable.getAnnotations()) {
+            String annotationName = annotation.getElementName();
+            if (annotationName.startsWith(JsonbConstants.JSONB_PREFIX) 
+                    && !annotationName.equals(JsonbConstants.JSONB_TRANSIENT)) {
+                return true;
+            }
+        }
+        return false;
+    }
+    
+    private boolean isAccessorOf(AccessorType type, IMethod method, IField field) {
+        String fieldName = field.getElementName();
+        String accessorName = type.name().toLowerCase() + fieldName.substring(0, 1).toUpperCase() + fieldName.substring(1);
+        return method.getElementName().equals(accessorName);
+    }
+    
+    private enum AccessorType {
+        GET, SET
```

Comment:
consider moving these to the Constants class

Reply:
Since the `isAccessor` method is moved to JDTUtils and there is not a common constants class, I also moved the enum to JDTUtils as it is used alongside `isAccessor`. But let me know if there is a better way.